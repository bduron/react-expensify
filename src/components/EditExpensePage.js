import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { startRemoveExpense } from '../actions/expenses';
import { editExpense } from '../actions/expenses';

export class EditExpensePage extends React.Component {
	onSubmit = (expense) => {
		this.props.editExpense(this.props.expense.id, expense);
		this.props.history.push('/');
	}

	onRemove = (e) => {
		this.props.startRemoveExpense({ id: this.props.expense.id});	
		this.props.history.push('/');	
	}

	render() {
		return (
			<div>
				<ExpenseForm 
					expense={this.props.expense}
					onSubmit={this.onSubmit}
				/>
				<button onClick={this.onRemove}>remove</button>
			</div>
		);
	}	
}

const mapDispatchToProps = (dispatch) => ({
	startRemoveExpense: (id) => dispatch(startRemoveExpense(id)),
	editExpense: (id, expense) => dispatch(editExpense(id, expense))
});

const mapStateToProps = (state, props) => ({
	expense: state.expenses.find((expense) => props.match.params.id === expense.id)
});

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);
