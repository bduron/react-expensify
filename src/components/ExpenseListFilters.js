import React from 'react';
import { connect } from 'react-redux';
import { setTextFilter, setStartDate, setEndDate } from '../actions/filters';
import { sortByAmount, sortByDate } from '../actions/filters';
import 'react-dates/initialize';
import { DateRangePicker } from 'react-dates';

export class ExpenseListFilters extends React.Component { 

	state = {
		calendarFocused: null
	};

	onDateChange = ({ startDate, endDate }) => {
		this.props.setStartDate(startDate); 	
		this.props.setEndDate(endDate); 	
	};

	onFocusChange = (focusedInput) => {
		this.setState(() => ({ calendarFocused: focusedInput }) );
	};

	onTextChange = (e) => {
		this.props.setTextFilter(e.target.value);		
	};

	onSortChange = (e) => {
		let action = e.target.value;

		if (action === 'date')
			this.props.sortByDate(); 
		if (action === 'amount')
			this.props.sortByAmount(); 
	};

	render() {
		return (
			<div className="content-container">
				<div className="input-group">
					<div className="input-group__item">
						<input 
							className="text-input" 
							type="text" 
							placeholder="Search expenses"
							value={this.props.filters.text} 
							onChange={this.onTextChange} 
						/>
					</div>
					<div className="input-group__item">
						<select 
							className="select"
							value={this.props.filters.sortBy}
							onChange={this.onSortChange}>
						<option value="date">Date</option>
						<option value="amount">Amount</option>
						</select>
					</div>
					<div className="input-group__item">
						<DateRangePicker
							startDate={this.props.filters.startDate} 
							endDate={this.props.filters.endDate} 
							onDatesChange={this.onDateChange} 
							focusedInput={this.state.calendarFocused} 
							onFocusChange={this.onFocusChange}
							numberOfMonths={1}
							isOutsideRange={() => false}
							showClearDates={true}
						/>
					</div>
				</div>
			</div>
		);
	}
}

const mapDispatchToProps = (dispatch) => ({
	setStartDate: (startDate) => dispatch(setStartDate(startDate)),
	setEndDate: (endDate) => dispatch(setEndDate(endDate)),
	setTextFilter: (char) => dispatch(setTextFilter(char)),
	sortByDate: () => dispatch(sortByDate()),
	sortByAmount: () => dispatch(sortByAmount())
});

const mapStateToProps = (state) => ({
	filters: state.filters
});

export default connect(mapStateToProps, mapDispatchToProps)(ExpenseListFilters);
