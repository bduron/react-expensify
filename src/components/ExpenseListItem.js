import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';

const ExpenseListItem = ({ id, description, amount, createdAt }) => (
	<div>
		<li>
			<b>{description}</b> : {amount}$ | {createdAt} | <Link to={"/edit/" +  id}>Edit </Link>
		</li>
	</div>
);

export default ExpenseListItem;
