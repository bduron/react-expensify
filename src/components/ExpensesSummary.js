import React from 'react';
import { connect } from 'react-redux';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';
import numeral from 'numeral';

export const ExpensesSummary = ({ expensesCount, expensesTotal }) => {
	const expenseWord = expensesCount === 1 ? 'expense' : 'expenses';
	const formattedExpensesTotal = numeral(expensesTotal / 100).format('$0,0.00');

	return (
		<div>
			<h2>Viewing {expensesCount} {expenseWord} totalling {formattedExpensesTotal}</h2>
		</div>
	)
};

const mapStateToProps = (state) => {
	const expenses = selectExpenses(state.expenses, state.filters);

	return {
		expensesCount: expenses.length,
		expensesTotal: selectExpensesTotal(expenses)
	};
};

export default connect(mapStateToProps)(ExpensesSummary); 
