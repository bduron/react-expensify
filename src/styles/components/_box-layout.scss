.box-layout {
	background: #FFFFFF; 
	//background: linear-gradient(to left, #485563, #29323c);
	display: flex;
	align-items: center;
	justify-content: center;
	height: 100vh;
	width: 100vw;
}

.box-layout__box {
	background: fade-out(white, .15);
	border-radius: 3px;
	text-align: center;
	width: 25rem;
	padding: $l-size $m-size;
}

.box-layout__title {
	margin: 0 0 $m-size 0; 
	line-height: 1;
}
