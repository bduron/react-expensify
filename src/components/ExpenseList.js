import React from 'react';
import { connect } from 'react-redux';
import ExpenseListItem from './ExpenseListItem'
import selectExpenses from '../selectors/expenses';

export const ExpenseList = (props) => (
	<div>
		{
			props.expenses.length !== 0 ? (
				props.expenses.map((expense) => <ExpenseListItem key={expense.id} {...expense} />)
			) : (
				<p>No expense to display</p>
			) 
		}
	</div>
);

const mapStateToProps = (state) => ({
	expenses: selectExpenses(state.expenses, state.filters)
});

export default connect(mapStateToProps)(ExpenseList); 
