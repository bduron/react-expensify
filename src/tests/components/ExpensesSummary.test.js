import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test('should render ExpensesSummary correctly - single expense', () => {
	const wrapper = shallow(<ExpensesSummary expensesCount={1} expensesTotal={2599} />);
	expect(wrapper).toMatchSnapshot();
});

test('should render ExpensesSummary correctly - multiple expenses', () => {
	const wrapper = shallow(<ExpensesSummary expensesCount={99} expensesTotal={259912938} />);
	expect(wrapper).toMatchSnapshot();
});
