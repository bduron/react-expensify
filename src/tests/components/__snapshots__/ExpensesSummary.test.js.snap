// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary correctly - multiple expenses 1`] = `
<div>
  <h2>
    Viewing 
    99
     
    expenses
     totalling 
    $2,599,129.38
  </h2>
</div>
`;

exports[`should render ExpensesSummary correctly - single expense 1`] = `
<div>
  <h2>
    Viewing 
    1
     
    expense
     totalling 
    $25.99
  </h2>
</div>
`;
